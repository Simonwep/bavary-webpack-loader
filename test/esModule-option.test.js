import {compile, execute, getCompiler, readAsset} from './helper';

describe('Option: "esModule"', () => {
    it('Should work without value', async () => {
        const compiler = getCompiler('simple/index.js');
        const stats = await compile(compiler);

        expect(execute(readAsset('main.bundle.js', compiler, stats))).toMatchSnapshot('result');
        expect(stats.compilation.warnings).toMatchSnapshot('warnings');
        expect(stats.compilation.errors).toMatchSnapshot('errors');
    });

    it('Should work with "Boolean" value equal "true"', async () => {
        const compiler = getCompiler('simple/index.js', {
            esModule: true
        });

        const stats = await compile(compiler);
        expect(execute(readAsset('main.bundle.js', compiler, stats))).toMatchSnapshot('result');
        expect(stats.compilation.warnings).toMatchSnapshot('warnings');
        expect(stats.compilation.errors).toMatchSnapshot('errors');
    });

    it('Should work with "Boolean" value equal "false"', async () => {
        const compiler = getCompiler('simple/index.js', {
            esModule: false
        });

        const stats = await compile(compiler);
        expect(execute(readAsset('main.bundle.js', compiler, stats))).toMatchSnapshot('result');
        expect(stats.compilation.warnings).toMatchSnapshot('warnings');
        expect(stats.compilation.errors).toMatchSnapshot('errors');
    });
});
