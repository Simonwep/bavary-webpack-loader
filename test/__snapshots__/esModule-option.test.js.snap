// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Option: "esModule" Should work with "Boolean" value equal "false": errors 1`] = `Array []`;

exports[`Option: "esModule" Should work with "Boolean" value equal "false": result 1`] = `"A"`;

exports[`Option: "esModule" Should work with "Boolean" value equal "false": warnings 1`] = `Array []`;

exports[`Option: "esModule" Should work with "Boolean" value equal "true": errors 1`] = `Array []`;

exports[`Option: "esModule" Should work with "Boolean" value equal "true": result 1`] = `"A"`;

exports[`Option: "esModule" Should work with "Boolean" value equal "true": warnings 1`] = `Array []`;

exports[`Option: "esModule" Should work without value: errors 1`] = `Array []`;

exports[`Option: "esModule" Should work without value: result 1`] = `"A"`;

exports[`Option: "esModule" Should work without value: warnings 1`] = `Array []`;
